File: src/gen-lexeduct-browser.py
#!/usr/bin/env python

# Written in Python right now, but should be converted to node for consistency.

import os

def main():
    print "module = {};"
    print "makeFilter = {};"
    for filename in os.listdir('filter'):
        if not filename.endswith('.js'):
            continue
        filter_name = filename[:-3]
        print
        with open(os.path.join('filter', filename), 'r') as f:
            print f.read()
        print "makeFilter['%s'] = module.exports.makeFilter;" % filter_name


if __name__ == '__main__':
    main()

File: src/gen-lexeduct-browser.js
#!/usr/bin/env node

fs = require('fs');

function write(s) {
    process.stdout.write(s + "\n");
}

write("module = {};");
write("makeFilter = {};");
var files = fs.readdirSync('filter');
for (var i = 0; i < files.length; i++) {
    var filename = files[i];
    var filterName = filename.split('.js')[0];
    var text = fs.readFileSync('filter/' + filename);
    write(text);
    write("makeFilter['" + filterName + "'] = module.exports.makeFilter;");
}
